src/main/resources/script.sh

```diff
@@ -0,0 +1,1353 @@
+        exit 1
+    fi
+}
+
+uploadGirderFile() {
+    local URI=$1
+
+    local fileName=$(echo $URI | sed -r 's#^girder:/(//)?(.*/)?([^/].*)\?.*$#\3#i')
+    local apiUrl=$(echo $URI | sed -r 's/^.*[?&]apiurl=([^&]*)(&.*)?$/\1/i')
+    local fileId=$(echo $URI | sed -r 's/^.*[?&]fileid=([^&]*)(&.*)?$/\1/i')
+    local token=$(echo $URI | sed -r 's/^.*[?&]token=([^&]*)(&.*)?$/\1/i')
+
+    if [ ! $(which girder-client) ]; then
+        pip install --user girder-client
+        if [ $? != 0 ]; then
+            error "girder-client not in PATH, and an error occured while trying to install it."
+            error "Exiting with return value 1"
+            exit 1
+        fi
+    fi
+
+    COMMLINE="girder-client --api-url ${apiUrl} --token ${token} upload --parent-type folder ${fileId} ./${fileName}"
+    echo "uploadGirderFile, command line is ${COMMLINE}"
+    ${COMMLINE}
+    if [ $? != 0 ]; then
+        error "Error while uploading girder file"
+        error "Exiting with return value 1"
+        exit 1
+    fi
+}
+
+function upload {
+
+    local URI=$1
+    local ID=$2
+    local NREP=$3
+    local TEST=$4
+
+    startLog file_upload uri="${URI}"
+
+        if [[ ${URI} == shanoir:/* ]]
+    then
+        if [ "${TEST}" != "true" ]
+        then
+            if [[ "$REFRESHING_JOB_STARTED" == false ]]; then
+                                refresh_token ${URI} & 
+                REFRESH_PID=$!  
+                REFRESHING_JOB_STARTED=true
+            fi
+            uploadShanoirFile ${URI}
+        fi
+    elif [[ ${URI} == girder:/* ]]
+    then
+        if [ "${TEST}" != "true" ]
+        then
+            uploadGirderFile ${URI}
+        fi
+    elif [[ ${URI} == file:/* ]]
+    then
+        local FILENAME=`echo $URI | sed 's%file://*%/%'`
+        local NAME=`basename ${FILENAME}`
+
+        if [ -e $FILENAME ]
+        then
+            error "Result file already exists: $FILENAME"
+            error "Exiting with return value 1"
+            exit 1
+        fi
+
+        if [ "${TEST}" = "true" ]
+        then
+            echo "test result" > ${NAME}
+        fi
+
+        \mv $NAME $FILENAME
+        if [ $? != 0 ]
+        then
+            error "Error while moving result local file."
+            error "Exiting with return value 1"
+            exit 1
+        fi
+    else
+                local LFN=`echo "${URI}" | sed -r 's%^\w+://[^/]*(/[^?]+)(\?.*)?$%\1%'`
+        local NAME=${LFN##*/}
+
+        if [ "${TEST}" = "true" ]
+        then
+            LFN=${LFN}-uploadTest
+            echo "test result" > ${NAME}
+        fi
+
+        uploadLfnFile ${LFN} ${PWD}/${NAME} ${NREP}
+
+        if [ "${TEST}" = "true" ]
+        then
+            \rm -f ${NAME}
+        fi
+    fi
+    stopLog file_upload
+}
+
+
+function delete {
+
+    local URI="$1"
+    local TEST="$2"
+
+    startLog file_delete uri="${URI}"
+
+    ## The pattern must NOT be put between quotation marks.
+    if [[ ${URI} == girder:* ]]; then
```

Comment:
missing "/" in regex, have you changed on purpose ?

Reply:
put back